src/explorer/trialApp/TrialAppTreeItem.ts

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { RequestError } from 'request-promise/errors';
+import { IActionContext, TreeItemIconPath } from 'vscode-azureextensionui';
+import { localize } from '../../localize';
+import { openUrl } from '../../utils/openUrl';
+import { getIconPath } from '../../utils/pathUtils';
+import { requestUtils } from '../../utils/requestUtils';
+import { AzureAccountTreeItem } from '../AzureAccountTreeItem';
+import { ISiteTreeItem } from '../ISiteTreeItem';
+import { SiteTreeItemBase } from '../SiteTreeItemBase';
+import { ITrialAppMetadata } from './ITrialAppMetadata';
+
+export class TrialAppTreeItem extends SiteTreeItemBase implements ISiteTreeItem {
+    public static contextValue: string = 'trialApp';
+    public contextValue: string = TrialAppTreeItem.contextValue;
+
+    public metadata: ITrialAppMetadata;
+
+    public defaultHostName: string;
+
+    public defaultHostUrl: string;
+
+    private readonly _defaultHostName: string;
+
+    public get iconPath(): TreeItemIconPath {
```

Comment:
You should be able to move this into `SiteTreeItemBase` with the idea that this is a shared UI element.  But leave the overloaded `get iconPath` in `DeploymentSlotTreeItem`